demo/api/src/db/fixtures/fixtures.console.ts

```diff
@@ -73,130 +66,30 @@ export class FixturesConsole {
-        );
-        pageTreeNodes.test3 = node;
+        const node = await this.pageGeneratorService.generatePage({ name: "Home", scope });
+        await this.pageGeneratorService.generatePage({ name: "Sub", scope, parentId: node.id });
 
-        await this.pageTreeService.updateNodeVisibility(node.id, PageTreeNodeVisibility.Published);
+        for (let i = 0; i < 3; i++) {
+            const page = await this.pageGeneratorService.generatePage({ name: `Page ${i}`, scope });
 
-        node = await this.pageTreeService.createNode(
-            {
-                name: "Link1",
-                slug: "link",
-                attachedDocument: {
-                    id: attachedDocumentIds[3],
-                    type: "Link",
-                },
-            },
-            PageTreeNodeCategory.MainNavigation,
-            scope,
-        );
-        pageTreeNodes.link1 = node;
-
-        await this.pageTreeService.updateNodeVisibility(node.id, PageTreeNodeVisibility.Published);
-
-        node = await this.pageTreeService.createNode(
-            {
-                name: "Test Site visibility",
-                slug: "test-site-visibility",
-                attachedDocument: {
-                    id: attachedDocumentIds[4],
-                    type: "Page",
-                },
-            },
-            PageTreeNodeCategory.MainNavigation,
-            scope,
-        );
-        pageTreeNodes.testSiteVisibility = node;
-
-        await this.pageTreeService.updateNodeVisibility(node.id, PageTreeNodeVisibility.Published);
-
-        for (const attachedDocumentId of attachedDocumentIds) {
-            const pageInput = getDefaultPageInput();
-
-            await this.pagesRepository.persistAndFlush(
-                this.pagesRepository.create({
-                    id: attachedDocumentId,
-                    createdAt: new Date(),
-                    updatedAt: new Date(),
-                    content: pageInput.content.transformToBlockData(),
-                    seo: pageInput.seo.transformToBlockData(),
-                }),
-            );
+            for (let subPageIndex = 0; subPageIndex < 3; subPageIndex++) {
+                if (faker.datatype.boolean()) {
+                    await this.pageGeneratorService.generatePage({ name: `Sub-Page ${subPageIndex}`, scope, parentId: page.id });
+                } else {
+                    await this.linkGeneratorService.generateLink({ name: `Sub-Link ${subPageIndex}`, scope, parentId: page.id });
+                }
+            }
         }
 
-        console.log("generate links");
-        await generateLinks(this.linksRepository, pageTreeNodes);
-        console.log("links generated");
-
-        console.log("generate many images test page");
-        const manyImagesTestGenerator = new ManyImagesTestPageGenerator(this.pageTreeService, this.filesService, this.pagesRepository);
-        await manyImagesTestGenerator.execute();
-        console.log("many images test page created");
-
```

Comment:
I'm not sure if we should delete this

Reply:
generateLinks:
 - was deleted because links are now generated in the linkGeneratorService

manyImagesTestGenerator:
this just generates a page with imageBlocks, is that useful?
Should that be needed I'd extend the pageGeneratorService but not just for the ImageBlock

The PageGeneratorService could: 
 - support an option: which block or blocks should be generated
 - support an option: how many blocks should be generated

Being able to generate specific blocks might be useful in my opinion, so that a page for every blockCategory can be generated or even a page for every block. In my opinion that's useful for bigger projects or when certain blocks are only allowed in certain menu types (main menu, top menu), but that's something I would have to do after the blocks-rework. (Since I'd have to adapt the page-content.generator as well)

Reply:
Okay, then we'll re-add this once you've refactored the block generators 🙂